shared/src/arguments.rs

```diff
@@ -49,6 +49,9 @@ pub struct Arguments {
     )]
     pub gas_estimators: Vec<GasEstimatorType>,
 
+    #[structopt(long, env)]
+    pub gas_estimator_api_key: Vec<String>,
```

Comment:
Should this be optional? If it is not set, then we could fall back on whatever we were doing before without this api key? Or possibly have a default value that aligns with the default of `gas_estimators` above. 

Actually, now that I think of it, I would propose that `gas_estimators` be upgraded to a vector of tuples `Vec<(GasEstimatorType, ApiKey)>`. So that we don't have to assert their lengths are equal.

Reply:
Imo this should be` blocknative_api_key` . If `GasEstimatorType` includes `BlockNative` we will ensure in `main` that the api key is set. I prefer this over the tuple or vec because not every estimator needs an api key.

Reply:
Changed to `pub blocknative_api_key: Option<String>`